testsuite/test_serverclass.py

```diff
@@ -0,0 +1,267 @@
+        include file since the compiler finds the files for qualifiers
+        and dependent classes.
+        TODO: ks 3/18 Right now we must build a file in the schema directory
+        to do this because the compile_from_string does not support
+        include files, etc. It does not search for files. See issue 1138
+        """
+        install_dmtf_schema()
+        class_list = """
+            #pragma locale ("en_US")
+            #pragma include ("Interop/CIM_RegisteredProfile.mof")
+            #pragma include ("Interop/CIM_Namespace.mof")
+            #pragma include ("Interop/CIM_ObjectManager.mof")
+            """
+        # See issue 1138.
+        # conn.compile_mof_str(class_list, namespace=namespace,
+        #                      search_paths=SCHEMA_MOF_DIR)
+        with open(TEST_SCHEMA, "w") as schema_file:
+            schema_file.write(class_list)
+
+        conn.compile_mof_file(TEST_SCHEMA, namespace=namespace,
+                              search_paths=[SCHEMA_MOF_DIR])
+
+        if os.path.isfile(TEST_SCHEMA):
+            os.remove(TEST_SCHEMA)
+        return
+
+    def inst_from_class(self, conn, cls_name, namespace=None, PropertyList=None,
+                        PropertyValues=None, IncludeNullProperties=True,
+                        strict=False, IncludePath=True):
+        """
+        Build an instance from the input classname.
+
+        Parameters:
+          conn (:class:`~pywbem.WBEMConnection`):
+            The WBEMConnection used to get the class from which
+            properties are derived for the instance.
+
+          cls_name
+            String with the class name from which properties are to
+            be derived for the instance.
+
+          namespace (:term:`string`):
+            Namespace in the WBEMConnection used to retrieve the class or
+            `None` if the default_namespace is to be used.
+
+          PropertyList (string or iterable of string)
+            Limits the properties retrieved witht the class and therefore
+            the properties in the resulting instance.
+
+          PropertyValues (dictionary):
+            Dictionary containing nane/value pairs where the names are the
+            names of properties in the class and the properties are the
+            property values to be set into the instance.
+
+          IncludeNullProperties (:class:`py:bool`:):
+            Determines if properties with Null values are included in the
+            instance.
+
+          strict (:class:`py:bool`:):
+            If `True` the property type in PropertyValue must exactly match
+            the property type in the class. Also, if IncludePath is `True` the
+            created instance is tested to assure that all key properties from
+            the class are in the instance so a complete path can be buil
+
+            If not `True` the value is inserted into the property whatever the
+            type
```

Comment:
non-strict apparently provides for the capability to set up mocked instances with property values that have CIM types that do not match the CIM types declared in the class. Is that the intention?

Reply:
Yes but I want to rethink that before we would consider moving this to the pywbem code.  I think I probably should build the CIMProperty with the value so that all the attributes of construction of a CIMProperty are exercised.  that would make use of CIMProperty complete construction tools and eliminate the the test I have which requires an exact match.  The current code works for the test but not if we were to move this to pywbem

Reply:
Modified the code so that strict only applies to the path now.  It was my error that I was trying to double check what CIMInstance.value did.